sklearn/metrics/_classification.py

```diff
@@ -2949,16 +2920,22 @@ def log_loss(
             1 - transformed_labels, transformed_labels, axis=1
         )
 
-    # Clipping
-    y_pred = np.clip(y_pred, eps, 1 - eps)
-
     # If y_pred is of single dimension, assume y_true to be binary
     # and then check.
     if y_pred.ndim == 1:
         y_pred = y_pred[:, np.newaxis]
     if y_pred.shape[1] == 1:
         y_pred = np.append(1 - y_pred, y_pred, axis=1)
 
+    # Make sure y_pred is normalized
+    y_pred_sum = y_pred.sum(axis=1)
+    if not np.allclose(y_pred_sum, 1):
+        raise ValueError("The y_pred values do not sum to one.")
+
+    # Clipping
+    eps = np.finfo(y_pred.dtype).eps
```

Comment:
Could we mention this clipping in the docstring?

Reply:
done